### DISCUSSION TITLE
How does renovate dependency on the github-api work ?

### DISCUSSION THREAD

**Discussion by kiranpatel11** [renovatebot/renovate#23049] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

We are running renovate in a self-hosted gitlab pipeline using the latest renovate container image. Our environment is restricted and does not have internet connectivity due to security reasons, hence cannot connect to the GITHUB API. We do have private registries which has whitelisted public repository access via the proxy cache.

Would like to understand more about the GitHub API invocation that renovate depend on (I believe it is for release notes at high level).
- What GitHub api and data being used via the GITHUB_TOKEN?
- does the data belong to the multiple repositories ? 
- who owns/contribute to such data ?
- Is there a possibility to host such data in-house and periodically sync-up with the upstream, potentially after some validations ?

### What you tried so far.

I went through the documentation but could not find details on how it really works. any help on these questions would really be appreciated.

I even tried disabling the github-api calls but it did not work out.

### Relevant debug logs

N/A

**Comment by rarkins** [comment]
You are mixing gitlab and github in your description. Can you edit to fix the typos?

**Reply by kiranpatel11** [reply]
thanks for pointing it out, my fingers have developed a muscle memory after working on gitlab for a while ;-)

**Comment by rarkins** [comment, marked as answer]
> * What GitHub api and data being used via the GITHUB_TOKEN?

I think you mean GITHUB_COM_TOKEN.

This is used both for querying releases/tags (e.g. for Node.js updates) but mostly for release notes, where it's used to query release content plus file content (e.g. `CHANGELOG.md`).

> * does the data belong to the multiple repositories ?

Yes

> * who owns/contribute to such data ?

The respective repositories/packages

> * Is there a possibility to host such data in-house and periodically sync-up with the upstream, potentially after some validations ?

No

**Reply by kiranpatel11** [reply]
can you mention the exact api urls being called so that we can try to get it whitelisted in our proxy ?

**Reply by rarkins** [reply]
That's impossible because it depends on what dependencies you have.

**Reply by kiranpatel11** [reply]
ohk

**Reply by viceice** [reply]
you should simply run the full renovate image and check the debug logs. for testing you can use the dry run mode. 🤷‍♂️
